src/pages/SignUp/SignUp.jsx

```diff
@@ -0,0 +1,175 @@
+          }
+        })
+        .catch(function (error) {
+          console.log(error);
+        });
+    },
+  });
+
+  const checkDisableJoinButton = () => {
+    return !isAgree;
+  };
+
+  return (
+    <Grid container direction='row' justifyContent='center' alignItems='center'>
+      <TermsDialog open={open} setOpen={setOpen} setIsAgree={setIsAgree} />
+      <Container maxWidth='sm'>
+        <Box
+          sx={{
+            margin: 8,
+            display: 'flex',
+            flexDirection: 'column',
+            alignItems: 'center',
+          }}
+        >
+          <Avatar />
+          <Typography component='h1' variant='h5'>
+            {t('Sign up')}
+          </Typography>
+          <Box component='form' onSubmit={formik.handleSubmit}>
+            <TextField
+              margin='normal'
+              fullWidth
+              autoFocus
+              label={t('Email Address')}
+              id='email'
+              name='email'
+              onChange={formik.handleChange}
+              value={formik.values.email}
+            />
+            {formik.touched.email && formik.errors.email ? (
+              <div style={errorStyle}>{formik.errors.email}</div>
+            ) : null}
+            <TextField
+              margin='normal'
+              fullWidth
+              autoFocus
+              label={t('Username')}
+              id='username'
+              name='username'
+              onChange={formik.handleChange}
+              value={formik.values.username}
+            />
+            {formik.touched.username && formik.errors.username ? (
+              <div style={errorStyle}>{formik.errors.username}</div>
+            ) : null}
+            <TextField
+              margin='normal'
+              fullWidth
+              autoFocus
+              label={t('Password')}
+              id='password'
+              type='password'
+              name='password'
+              onChange={formik.handleChange}
+              value={formik.values.password}
+            />
+            {formik.touched.password && formik.errors.password ? (
+              <div style={errorStyle}>{formik.errors.password}</div>
+            ) : null}
+            <TextField
+              margin='normal'
+              fullWidth
+              autoFocus
+              label={t('Confirm password')}
+              id='confirmPassword'
+              type='password'
+              name='confirmPassword'
+              onChange={formik.handleChange}
+              value={formik.values.confirmPassword}
+            />
+            {formik.touched.confirmPassword && formik.errors.confirmPassword ? (
+              <div style={errorStyle}>{formik.errors.confirmPassword}</div>
+            ) : null}
+            <FormControlLabel
+              control={
+                <Checkbox
+                  onClick={() => setOpen(true)}
+                  value='term'
+                  color='primary'
+                />
```

Comment:
This checkbox must be controlled, more precisely the value of the checkbox must reflect what we have in the state `formik.values.acceptTerms`.
```
                <Checkbox
                  onClick={() => setOpen(true)}
                  value='term'
                  color='primary'
                  checked={formik.values.acceptTerms}
                />
```